UefiCpuPkg/PiSmmCpuDxeSmm/SmmProfile.c

```diff
@@ -490,6 +497,18 @@ InitProtectedMemRange (
         continue;
       }
 
+      if (((MemorySpaceMap[Index].BaseAddress & (SIZE_4KB - 1)) != 0) ||
+          ((MemorySpaceMap[Index].Length & (SIZE_4KB - 1)) != 0))
+      {

```

Comment:
I agree that if-check is needed here. But do you think that moving warning to above is better?

Reply:
Moved the warning to above